backend/middlewares/validators/util.ts

```diff
@@ -39,6 +39,26 @@ export const validateArray = (value: any, type: Type): boolean => {
   );
 };
 
+export const validateObject = (
+  object: { [field: string]: any },
+  types: { [field: string]: any },
+): boolean => {
+  return (
+    object !== undefined &&
+    object !== null &&
+    typeof object === "object" &&
+    Object.entries(object).every(([key, value]) => {
+      if (key.slice(-2) === "[]" && Array.isArray(value)) {
+        return validateArray(value, types[key.slice(0, -2)]);
+      }
```

Comment:
Bit unclear on this piece of logic, what's happening here?

Reply:
Basically, `string[]` represents an array, so if the last 2 chars of a string are '[]', then we should use validate array. When passing into validateArray, we only want the type of each item, so we remove the `[]` characters.

Reply:
Makes sense, maybe worth a comment